test/Microsoft.AspNetCore.Mvc.FunctionalTests/compiler/resources/HtmlGenerationWebSite.HtmlGeneration_Home.Index.html

```diff
@@ -39,35 +39,13 @@
         </a>
     </div>
     <div>
-        <a href="/Customer/Customer#fragment">Customer Area Customer Index</a>
+        <a href="/Customer/HtmlGeneration_Customer#fragment">Customer Area Customer Index</a>
     </div>
     <div>
         <a href="/Order/List">Href Order List</a>
     </div>
-    <div>
-        <a href="/NonExistentController">Non-existent Controller</a>
```

Comment:
I don't feel strongly that these need to be tested. We would be generating links which cannot be reached at all.

Reply:
I am not sure how we can achieve this with the way things are setup currently. These kind of links can be generated only by parameterized templates and we do not have them by default as we expand(or denormalize) them in MvcEndpointDataSource.
If we were to add the parameterized templates as fallback endpoints (at the end of the list), then the invoker for these needs to be different now as we need to do something that is similar to pre-dispatcher world. Let's discuss more.